models/basgra/R/run_BASGRA.R

```diff
@@ -0,0 +1,268 @@
+##' @param start_date start time of the simulation
+##' @param end_date end time of the simulation
+##' @param outdir where to write BASGRA output
+##' @param sitelat latitude of the site
+##' @param sitelon longitude of the site
+##' 
+##' @export
+##' @useDynLib PEcAn.BASGRA
+##' @author Istem Fer
+##-------------------------------------------------------------------------------------------------#
+
+run_BASGRA <- function(run_met, run_params, start_date, end_date, outdir, sitelat, sitelon){
+
+  start_date  <- as.POSIXlt(start_date, tz = "UTC")
+  end_date    <- as.POSIXlt(end_date, tz = "UTC")
+  start_year  <- lubridate::year(start_date)
+  end_year    <- lubridate::year(end_date)
+  
+
+
+  ################################################################################
+  ### FUNCTIONS FOR READING WEATHER DATA
+  mini_met2model_BASGRA <- function(file_path,
+                                    start_date, start_year,
+                                    end_date, end_year) {
+
+    out.list <- list()
+    
+    ctr <- 1
+    for(year in start_year:end_year) {
+      
+      diy <- PEcAn.utils::days_in_year(year)
+      
+      NWEATHER              <- as.integer(8)
+      matrix_weather        <- matrix( 0., nrow=diy, ncol=NWEATHER )
+      
+      
+      # prepare data frame for BASGRA format, daily inputs, but doesn't have to be full year
+      
+      
+      matrix_weather[ ,1] <- rep(year, diy) # year
+      matrix_weather[ ,2] <- seq_len(diy) # day of year, simple implementation for now
+      
+      old.file <- file.path(dirname(file_path), paste(basename(file_path), year, "nc", sep = "."))
+      
+      if (file.exists(old.file)) {
+        
+        ## open netcdf
+        nc <- ncdf4::nc_open(old.file)  
+        
+        ## convert time to seconds
+        sec <- nc$dim$time$vals
+        sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
+        
+        dt <- PEcAn.utils::seconds_in_year(year) / length(sec)
+        tstep <- round(86400 / dt)
+        dt <- 86400 / tstep
+        
+        ind <- rep(seq_len(diy), each = tstep)
+        
+        rad <- ncdf4::ncvar_get(nc, "surface_downwelling_shortwave_flux_in_air")
+        gr  <- rad *  0.0864 # W m-2 to MJ m-2 d-1
+        
+        matrix_weather[ ,3]  <- tapply(gr, ind, mean, na.rm = TRUE) # irradiation (MJ m-2 d-1)
+        
+        Tair <-ncdf4::ncvar_get(nc, "air_temperature")  ## in Kelvin
+        Tair_C <- udunits2::ud.convert(Tair, "K", "degC")
+        
+        t_dmean <- tapply(Tair_C, ind, mean, na.rm = TRUE) # maybe round these numbers 
+        matrix_weather[ ,4] <- t_dmean # mean temperature (degrees Celsius)
+        matrix_weather[ ,5] <- t_dmean # that's what they had in read_weather_Bioforsk
+        
+        RH <-ncdf4::ncvar_get(nc, "relative_humidity")  # %
+        RH <- tapply(RH, ind, mean, na.rm = TRUE) 
+        
+        matrix_weather[ ,6] <- exp(17.27*t_dmean/(t_dmean+239)) * 0.6108 * RH / 100
```

Comment:
Can you add a brief comment explaining what this calculation is? It looks vaguely like a conversion to specific humidity, but I'm not sure.

Reply:
I was following their calculation here https://github.com/MarcelVanOijen/BASGRA_N/blob/eafc6e36cf4df6ba04638b14a56d28616296fb97/initialisation/initialise_BASGRA_general.R#L40

Looking at [here](https://github.com/MarcelVanOijen/BASGRA_N/blob/eafc6e36cf4df6ba04638b14a56d28616296fb97/model/BASGRA.f90#L86) and `https://github.com/MarcelVanOijen/BASGRA_N/blob/eafc6e36cf4df6ba04638b14a56d28616296fb97/model/environment.f90#L49`, it is Vapor Pressure
```
VPI   = MATRIX_WEATHER(:,6)
...
VP     = VPI(day)   ! vapour pressure (kPa)
```

Reply:
Ah, OK. Thanks!